### ISSUE TITLE
Create a Human Readable Doc from the Config Schema

### ISSUE THREAD

**Issue by alyec** [fgpv-vpgf/fgpv-vpgf#1076]
### Current design and implementation progress
- [x] Setup [outside repo](https://github.com/milesap/schema-to-docs/tree/develop) with basic npm support 
- [x] **Stage 1 transformation:** Dereference `$ref` objects (using a schema.json copy from fgpv).
- [x] **Stage 2 transformation:** Circular references are ignored in stage one. Manually deference circular dependencies **once** and add an additional attribute to it for later identification.
- [x] **Stage 3 transformation:** Simplification -  this involves adding/removing attributes to help improve readability. For example, `allOf` would be replaced by all of its requirements. Similarly, `oneOf` is spawned into multiple possible combinations of actual implementation schema.
- [x] **Stage 4 transformation:** Convert into a JS object, which enforces any circular attributes which may cause issues on the following step.
- [x] **Rendering:** Given the JS object, render it as an interactive HTML document(s).
- [x] **Integration:** Test importing library in FGPV, schema passing and returning works as intended
- [ ] **Optional:** Allow support as a webpack plugin

Once this is complete, a new task will be assigned to go through the schema to correct any schema validation warnings (unused attributes), and to have each item well described.

### Initial issue post
From some basic testing at the time (2016) automated tools which take a JSON schema and produce docs run into issues.  Seems to be mostly an issue with following references.

Worth checking the existing tools to see if any of them produce reasonable documentation with our schema.

As an alternate approach we can use / write a tool which scraps every top level object, and then add some custom documentation showing how all the pieces link together (ideally an mspaint masterpiece from @james-rae ).

**Comment by milesap** [comment]
This is likely changing with refactors

**Comment by milesap** [comment]
FYI [JSON draft 7](http://json-schema.org/latest/json-schema-core.html) does not include inheritance, possibly added on draft 8 or never. [Draft vote to clode](https://github.com/json-schema-org/json-schema-spec/issues/15#issuecomment-347672090)

**Comment by milesap** [comment]
[Stage 2 Sample](https://raw.githubusercontent.com/milesap/schema-to-docs/0.0.3/drs/Stage%202.json)
[Stage 3 Sample](https://raw.githubusercontent.com/milesap/schema-to-docs/0.0.3/drs/Stage%203.json)

Need some :eyes: on the `Stage 3` config file above to ensure the schema remains well defined with no missing or added properties/side effects. `Stage 3` should be considered "more readable" to someone who doesn't know the original schema, and will help feed the document generator(s). `Stage 2` is provided if you want to see the dereferenced schema prior to simplification. 

Some definitions like `visibilitySet` and `entryGroup` are "removed", instead their contents/implementations are merged where appropriate. Disregard `#refTo` properties, these will be used on next two stages. 

I don't consider this "readable" to the average person yet but essentially I'll be sprinkling some HTML/CSS fairy dust over `Stage 3` next so that it becomes readable. Adding/updating content descriptions and quality will be handled in a separate issue.
